rakelib/contact_info_via_veteran_icn.rake

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+namespace :veteran_contact_info do
+  desc 'Build a CSV with contact information for a batch of veteran icns'
+  task :build_contact_info_csv, %i[csv_filename icn_list] => :environment do |_, args|
+    def build_address(mpi_address)
+      line1 = mpi_address.street
+      line2 = mpi_address.street2
+      street = line1
+      street += ", #{line2}" if line2.present?
+      city = mpi_address.city
+      postal_code = mpi_address.postal_code
+      country = mpi_address.country
+
+      "#{street}, #{city} #{postal_code}, #{country}"
+    end
+
+    def build_common_name(mpi_profile)
+      first_name = mpi_profile.given_names[0]
+      middle_name = mpi_profile.given_names[1]
+      last_name = mpi_profile.family_name
+      suffix = mpi_profile.suffix
+
+      [first_name, middle_name, last_name, suffix].compact.join(' ')
+    end
+
+    def retrieve_current_email(vet360_profile)
+      current_emails = vet360_profile.emails.select { |email| email.effective_end_date.nil? }
+
+      raise 'All emails expired!' if current_emails.length.zero?
+
+      current_emails.first.email_address
+    end
+
+    def upload_to_s3(_csv, csv_filename, csv_path, url_life_length: 1.week.to_i)
+      s3_settings = Settings.decision_review.s3
+
+      begin
+        s3_bucket = s3_settings.bucket
+        Rails.logger.info('Uploading contact info CSV to S3...')
+        s3_resource = Aws::S3::Resource.new(
+          region: s3_settings.region,
+          access_key_id: s3_settings.access_key_id,
+          secret_access_key: s3_settings.secret_access_key
+        )
+
+        obj = s3_resource.bucket(s3_bucket).object(csv_filename)
+        obj.upload_file(csv_path, content_type: 'text/csv')
+        obj.presigned_url(:get, expires_in: url_life_length)
```

Comment:
we ended up not having to use this code as the VA team (Cory on the enablement team) found a different way to retrieve the contact info, but i'm leaving some notes here in case someone in the future wants to repurpose this script

since we're running this code as a rake task, this presigned url doesn't get returned once the task finishes — it would have to be logged somehow. we were considering adding a success logger here and including the presigned url with a significantly shorter url life so that we can just grab it from the logs. 

other considerations include moving the CSV generation into a job and once the job was finished, and possibly running this upload code manually in a console or somehow retrieving the presigned url manually by looking up the resource. also want to note that testing this script against staging data didn't work (mock users not set up correctly or just in a messy state?) so there might need to be some massaging of data on there before you can test it there